plugins/modules/job_invocation.py

```diff
@@ -0,0 +1,223 @@
+      - ssh related options
+    type: dict
+    suboptions:
+      effective_user:
+      description:
+        - What user should be used to run the script (using sudo-like mechanisms)
+        - Defaults to a template parameter or global setting
+      type: str
+  command:
+    description:
+      - Command to be executed on host. Required for command templates
+    type: str
+  recurrence:
+    description:
+      - Schedule a recurring job
+    type: dict
+    suboptions:
+      cron_line:
+        description:
+          - How often the job should occur, in the cron format
+        type: str
+      max_iteration:
+        description:
+          - Repeat a maximum of N times
+        type: int
+      end_time:
+        description:
+          - Perform no more executions after this time
+        type: datetime
+  scheduling:
+    description:
+      - Schedule the job to start at a later time
+    type: dict
+    suboptions:
+      start_at:
+        description:
+          - Schedule the job for a future time
+        type: datetime
+      start_before:
+        description:
+          - Indicates that the action should be cancelled if it cannot be started before this time.
+        type: datetime
+  concurrency_control:
+    description:
+      - Control concurrency level and distribution over time
+    type: dict
+    suboptions:
+      time_span:
+        description:
+          - Distribute tasks over given number of seconds
+        type: int
+      concurrency_level:
+        description:
+          - Maximum jobs to be executed at once
+        type: int
+extends_documentation_fragment:
+  - theforeman.foreman.foreman
+  - theforeman.foreman.foreman.entity_state_with_defaults
+  - theforeman.foreman.foreman.taxonomy
+'''
+
+EXAMPLES = '''
+
+- name: "Run remote command on a single host once"
+  job_invocation:
+    search_query: "name ^ (foreman.example.com)"
+    command: 'ls'
+    job_template_id: 148
+    ssh:
+      - effective_user: "tester"
+
+- name: "Run ansible commad on active hosts once a day"
+  job_invocation:
+    bookmark_id: 6
+    command: 'pwd'
+    job_template_id: 177
+    recurrence:
+      - cron_line: "30 2 * * *"
+    concurrency_control:
+      - concurrency_level: 2
+'''
+
+RETURN = ''' # '''
+
+from ansible_collections.theforeman.foreman.plugins.module_utils.foreman_helper import (
+    ForemanEntityAnsibleModule,
+)
+
+ssh_foreman_spec = {
+    'effective_user': dict(),
+}
+
+recurrnece_foreman_spec = {
+    'cron_line': dict(),
+    'max_iteration': dict(),
+    'end_time': dict(),
+}
+
+scheduling_foreman_spec = {
+    'start_at': dict(),
+    'start_before': dict(),
+}
+
+concurrency_control_foreman_spec = {
+    'time_span': dict(),
+    'concurrency_level': dict(),
+}
+
+
+class ForemanJobInvocationModule(ForemanEntityAnsibleModule):
+    pass
+
+
+def main():
+    module = ForemanJobInvocationModule(
+        foreman_spec=dict(
+            search_query=dict(),
+            bookmark_id=dict(),
```

Comment:
This is where the `foreman_spec` magic starts (Have a look at the docs...).
I guess it will be something like:
```suggestion
            bookmark=dict(type='entity'),
```
however for bookmarks you will need to specify the controller, too.